tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/FindReplaceDialogTest.java

```diff
@@ -366,6 +378,37 @@ public void testFindWithWholeWordEnabledWithMultipleWords() {
 		assertEquals(dialog.findCombo.getText().length(), (target.getSelection()).y);
 	}
 
+	@Test
+	public void testGlobalSearchChangeScope() {
+		openTextViewerAndFindReplaceDialog("line1\nline2\nline3");
+
+		fTextViewer.setSelection(new TextSelection(6, 11));
+		dialog.globalRadioButton.setSelection(false);
+		dialog.selectedRangeRadioButton.setSelection(true);
+		dialog.selectedRangeRadioButton.notifyListeners(SWT.Selection, null);
```

Comment:
Please use the existing `select()` and `deselect()` operation instead of reimplementing the same logic.

Reply:
I have removed the test and added a similar test in FindReplaceLogicTest with this commit: https://github.com/eclipse-platform/eclipse.platform.ui/commit/fa4cfb1121b8ac0f2fe8f1bbb0ae46cc656dc22e